### PR TITLE
travis: Use the docker infrastructure

### PR DESCRIPTION
**Pull request by dawagner**
As advertised by travis news:

    http://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/

using the new docker infra is faster and more reliable. This comes with the
cost of not being able to use "sudo", though. Installing packages is still
possible by specifying them in an "addons.apt" section in .travis.yml. We have
to remove "sudo make install" but we aren't using the installed libs and
binaries anyway.

I had some syntax issues with the "addons" and "env" sections and had to remove
the coverity subsection which wasn't configure for the 01org repository anyway.
I'm also removing the email notification since I never received any email and
actually don't want to.

Signed-off-by: David Wagner <david.wagner@intel.com>

### CODE REVIEW

**Comment by dawagner** [comment]
@krocard please review

**Comment by gdenneul** [comment]
:+1:

File: .travis.yml [hunk 1]
```diff
@@ -1,41 +1,55 @@
...
+    global:
+        - PREFIX=$HOME/prefix
+
 compiler:
-  - gcc
-#  - clang # not supported yet
+    - gcc
+    - clang
+
+matrix:
```
Comment (krocard) [inline .travis.yml]: useless matrix, why did you not kept clang commented ?

File: .travis.yml [hunk 2]
```diff
@@ -1,41 +1,55 @@
...
+    allow_failures:
+        # clang not supported yet
+        - compiler: clang
```
Comment (krocard) [inline .travis.yml]: useless matrix, why did you not kept clang commented ?

File: .travis.yml [hunk 3]
```diff
@@ -1,41 +1,55 @@
...
 
 # Install a recent gcc, gcov and cmake,
```
Comment (krocard) [inline .travis.yml]: cmake are not longer updated.

File: .travis.yml [hunk 4]
```diff
@@ -1,41 +1,55 @@
...
-  - sudo apt-get install --yes swig cmake valgrind g++-4.8
-  - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 100
-  - sudo update-alternatives --install /usr/bin/gcov gcov /usr/bin/gcov-4.8 100
-  - sudo wget --directory-prefix /usr/include/
+    - pip install --user cpp-coveralls; export PATH=$HOME/.local/bin:$PATH
+    - wget --directory-prefix $PREFIX
               https://raw.github.com/philsquared/Catch/master/single_include/catch.hpp
 
+before_script:
+    - if [ "$CC" = "gcc" ]; then export CC=gcc-4.8 CXX=g++-4.8; fi
```
Comment (krocard) [inline .travis.yml]: export CC=${CC/$gcc$/gcc-4.8}
export CXX=${CC/$g++$/g++-4.8}

File: .travis.yml [hunk 5]
```diff
@@ -1,41 +1,55 @@
...
+            - valgrind
+            - g++-4.8
 
 install:
-  - sudo apt-get install --yes swig cmake valgrind g++-4.8
-  - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 100
-  - sudo update-alternatives --install /usr/bin/gcov gcov /usr/bin/gcov-4.8 100
-  - sudo wget --directory-prefix /usr/include/
+    - pip install --user cpp-coveralls; export PATH=$HOME/.local/bin:$PATH
+    - wget --directory-prefix $PREFIX
```
Comment (krocard) [inline .travis.yml]: > $ man cmake
> ***CMAKE_PREFIX_PATH***
>   Path used for searching by FIND_XXX(), with appropriate suffixes added.
> 
>   Specifies a path which will be used by the  FIND_XXX()  commands.  It  contains  the  "base"  directories,  the  FIND_XXX()  commands  append appropriate subdirectories to the base directories. So FIND_PROGRAM() adds /bin to each of the directories in the path, FIND_LIBRARY() appends /lib to each of the  directories,  and  FIND_PATH()  and  **FIND_FILE()  append  /include**  . By default it is empty, it is intended to be set by the project.

Catch is found with FIND_FILE. If I understand correctly the manual `cmake` looks for `catch.hpp` in `${CMAKE_PREFIX_PATH}/include`. Should you not download the header to `$PREFIX/include` ?

File: .travis.yml [hunk 6]
```diff
@@ -48,26 +62,9 @@ after_success:
           --exclude "test/test-subsystem"
           --exclude "bindings/c/Test.cpp"
           --exclude "test/tokenizer"
+          --gcov /usr/bin/gcov-4.8
           --gcov-options '\--long-file-names --preserve-paths'
 
 notifications:
-  email:
-    - david.wagner@intel.com
```
Comment (krocard) [inline .travis.yml]: How is this related to docker ?

File: .travis.yml [hunk 7]
```diff
@@ -48,26 +62,9 @@ after_success:
...
   irc:
     - "chat.freenode.net#parameter-framework"
-
```
Comment (krocard) [inline .travis.yml]: How is this change related to docker ?

File: .travis.yml (hunk 5 above)
Comment (dawagner) [inline .travis.yml]: oops, you are right; that's what I intended to do.

File: .travis.yml (hunk 7 above)
Comment (dawagner) [inline .travis.yml]: see commit message

File: .travis.yml (hunk 6 above)
Comment (dawagner) [inline .travis.yml]: see commit message... but you are right, in theory I should have done it in a separate patch.....

File: .travis.yml (hunk 3 above)
Comment (dawagner) [inline .travis.yml]: done

File: .travis.yml (hunk 2 above)
Comment (dawagner) [inline .travis.yml]: done

File: .travis.yml (hunk 6 above)
Comment (dawagner) [inline .travis.yml]: done

**Comment by krocard** [comment]
:+1:

File: .travis.yml [hunk 8]
```diff
@@ -1,41 +1,52 @@
...
+        cmake -DCMAKE_PREFIX_PATH=$PREFIX -DCMAKE_INSTALL_PREFIX=../install .. &&
         make -j &&
-        sudo make install &&
-        sudo ldconfig &&
-        CTEST_OUTPUT_ON_FAILURE=1 make test )
+        CTEST_OUTPUT_ON_FAILURE=1 make test &&
+        make install)
   - ( cd skeleton-subsystem &&
-        cmake . &&
+        cmake -DCMAKE_INSTALL_PREFIX=../install . &&
```
Comment (OznOg) [inline .travis.yml]: I usually prefer global path for this
